packages/cli/src/commands/setup/deploy/providers/render.js

```diff
@@ -0,0 +1,114 @@
+import fs from 'fs'
+import path from 'path'
+
+import { getPaths } from 'src/lib'
+
+const PROJECT_NAME = getPaths().base.match(/[^/|\\]+$/)[0]
+
+const RENDER_YAML = (database) => {
+  return `services:
+- type: web
+  name: ${PROJECT_NAME}-web
+  env: static
+  buildCommand: yarn rw deploy render web
+  staticPublishPath: ./web/dist
+  envVars:
+  - key: NODE_VERSION
+    value: 14
+  routes:
+  - type: rewrite
+    source: /.redwood/functions/*
+    destination: replace_me_with_api_url/*
+  - type: rewrite
+    source: /*
+    destination: /index.html
+
+- type: web
+  name: ${PROJECT_NAME}-api
+  env: node
+  buildCommand: yarn rw deploy render api
+  startCommand: cd api && yarn api-server --functions ./dist/functions
+  envVars:
+  - key: NODE_VERSION
+    value: 14
+${database}
+`
+}
+
+const POSTGRES_YAML = `  - key: DATABASE_URL
+  fromDatabase:
+    name: redwood-db
+    property: connectionString
+
+databases:
+  - name: ${PROJECT_NAME}-db
+`
+
+const SQLITE_YAML = `  - key: DATABASE_URL
+    value: file:./data/sqlite.db
+  disk:
+    name: sqlite-data
+    mountPath: /opt/render/project/src/api/db/data
+    sizeGB: 1`
+
+const RENDER_HEALTH_CHECK = `// render-health-check
+export const handler = async () => {
+  return {
+    statusCode: 200,
+  }
+}
+`
+// prisma data source check
+export const prismaDataSourceCheck = (database) => {
+  if (database === 'none') {
+    return {
+      path: path.join(getPaths().base, 'render.yaml'),
+      content: RENDER_YAML(''),
+    }
+  }
+  const content = fs.readFileSync(getPaths().api.dbSchema).toString()
+  const detectedDatabase = content.match(
+    /(?<=datasource DS.*\n\W*provider\W*)\w+/
+  )
+  if (detectedDatabase == database) {
+    switch (database) {
+      case 'postgres':
+        return {
+          path: path.join(getPaths().base, 'render.yaml'),
+          content: RENDER_YAML(POSTGRES_YAML),
+        }
+      case 'sqlite':
+        return {
+          path: path.join(getPaths().base, 'render.yaml'),
+          content: RENDER_YAML(SQLITE_YAML),
+        }
+      default:
+        throw new Error(`
+       Unexpected datasource provider found: ${database}`)
+    }
+  } else {
+    throw new Error(`
+    Prisma datasource provider is detected to be ${detectedDatabase}.
+    Update your schema.prisma provider to be ${database}, then run
+    yarn rw prisma migrate dev`)
+  }
+}
+
+//any packages to install
+export const apiPackages = ['@redwoodjs/api-server']
```

Comment:
```suggestion
```

Interesting, a few things here.
1. we should remove this as a default install -- we are now included this dependency in all new create-redwood-app installations
2. also, this installs as a devDependency. We now include as dependency, so it's installing a duplicate
3. however, in the case of existing, older projects, this package may not be installed. We should check and either a) warn or b) install if not found

@peterp Recommendations about how to resolve and/or update setup/deploy/deploy.js? The only other provider installing packages currently is aws-serverless ('@netlify/zip-it-and-ship-it' and 'serverless-dotenv-plugin'). Ideally in deploy.js we'd 1) check if package is installed in either dependencies or devDependencies and 2) perhaps provide option to install as devDependency or dependency.

Reply:
^^ also @dac09 (whoever comments first!)

Reply:
I think checking and asking to install a good solution for now. It would be awesome if our upgrade command was expanded enough to do such thing for us, since that feels like the natural place to do this.

Reply:
@SEANDOUGHTY is my explanation of the situation clear to you? If so, would you be able to take a stab at Peter's suggestion:
1. check to make sure @redwoodjs/api-server is installed in API package.json
2. if not, either run installation as dependency (preferred) or optionally warn user they need to install and give CLI instructions in console output

?

Reply:
I'll take a look!

Reply:
This is outdated given Sean's new code. See `/setup/deploy/deploy.js` comment starting L124